src/main/java/sparksoniq/spark/iterator/flowr/LetClauseSparkIterator.java

```diff
@@ -36,25 +38,113 @@
 import java.util.List;
 
 public class LetClauseSparkIterator extends FlowrClauseSparkIterator {
+
+    private VariableReferenceIterator _variableReference;
```

Comment:
I think we don't even need to store a VariableReferenceIterator anywhere. It is an overkill to pass an iterator just to tell the name of the let variable. You can already resolve this to a String in the constructor. In the future, we can even pass String variableName instead to the constructor (but no need to do it now).